lib/active_record_doctor/detectors/missing_unique_indexes.rb

```diff
@@ -116,6 +123,26 @@ def ignore_columns
           column.gsub(" ", "")
         end
       end
+
+      def expression_indexes_unsupported?
+        (ActiveRecord::VERSION::STRING < "5.0") ||
+          # Active Record < 6 is unable to correctly parse expression indexes for MySQL.
+          (mysql? && ActiveRecord::VERSION::STRING < "6.0")
+      end
+
+      def index_columns(columns)
+        return columns unless columns.is_a?(String)
```

Comment:
I'd like to make two points here:

1. Let's add a comment explaining why this is needed. What are types other than `String` that can be passed to the function.
2. Are there cases where this is an array and we'd need to perform the transformations below (those `gsub`s) on each of its elements?

Reply:
1. Added comment to the callsite
2. Seems like no. Rails returns columns as arrays only when they are "bare" columns. When one of them is an expression, the string is returned